inference-engine/tests/functional/plugin/myriad/subgraph_tests/topk_k_propagation.cpp

```diff
@@ -104,4 +111,45 @@ TEST_P(DynamicToStaticTopKPropagationConvert, KPropagation) {
 
 INSTANTIATE_TEST_CASE_P(smoke_NGraph, DynamicToStaticTopKPropagationConvert, ::testing::ValuesIn(kVec));
 
+class DynamicToStaticTopKPropagationShapeOfBased : public DynamicToStaticTopKPropagationConcatBased {
```

Comment:
The test class is named 'ShapeOfBased', but it has the default `buildSubgraph` method which builds subgraph with Gather.
I think such behavior is not applicable to `Based` class.

By the way, it looks strange that the `DynamicToStaticTopKPropagationShapeOfBased` is inherited from `DynamicToStaticTopKPropagationConcatBased` and don't use `concat`

I'd rename the class or add another inherited class related to Gather

Reply:
I've remade the classes, please take a look once more.